src/modules/mail/mail.service.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { MailerService } from '@nestjs-modules/mailer';
+import { ConfigService } from '@nestjs/config';
+
+import { ESubjectName } from './enums/subject-name.enum';
+import { EMailTemplate } from './enums/mail-template.enum';
+
+@Injectable()
+export class MailService {
+  constructor(
+    private readonly mailerService: MailerService,
+    private readonly configService: ConfigService,
+  ) {}
+  public async sendMail(
+    recipient: string,
+    subject: ESubjectName,
+    template: EMailTemplate,
+    context: Record<string, string | number> = {},
+  ): Promise<void> {
+    try {
+      await this.mailerService.sendMail({
+        to: recipient,
+        from: this.configService.get('MAIL_USER'),
+        subject,
+        template,
+        context,
+      });
+    } catch (error) {
+      if (
+        error instanceof HttpException &&
+        error.getStatus() === HttpStatus.BAD_REQUEST
+      ) {
+        throw error;
+      }
```

Comment:
I think you need create general nest error handler, to avoid wrapping each service to try/catch block. could be done in separate PR

Reply:
thats a question we want to ask You with our team)) cause writing try/catch everywhere is annoying so we have it just in case in feature/sign-in and will implement after Your approvement of this feature))